fuse_optimizers/src/fixed_lag_smoother.cpp

```diff
@@ -206,7 +207,24 @@ void FixedLagSmoother::optimizationLoop()
       // Combine the new transactions with any marginal transaction from the end of the last cycle
       new_transaction->merge(marginal_transaction_);
       // Update the graph
-      graph_->update(*new_transaction);
+      try
+      {
+        graph_->update(*new_transaction);
+      }
+      catch (const std::exception& ex)
+      {
+        std::ostringstream oss;
+        oss << "Graph:\n";
+        graph_->print(oss);
+        oss << "\nTransaction:\n";
+        new_transaction->print(oss);
+
+        ROS_FATAL_STREAM("Failed to update graph with transaction: " << ex.what()
```

Comment:
Is this really fatal? There is no way to recover?

Reply:
I could ignore the transaction and let it go, but I don't know how the graph or the transaction looked like when this happened.

I prefer to see how they look like the next time this happens, so I can see if there's an actual logic error that should be fixed, or otherwise something that could be ignored.

For context, this issue has happened only once, after a huge amount of runtime. I basically don't want to miss this the next time it happens, if ever.